docs/developer-guide/working-with-rules.md

```diff
@@ -139,6 +139,7 @@ Additionally, the `context` object has the following methods:
     * If the node is an `ImportSpecifier`, `ImportDefaultSpecifier`, or `ImportNamespaceSpecifier`, the declared variable is returned.
     * Otherwise, if the node does not declare any variables, an empty array is returned.
 * `getFilename()` - returns the filename associated with the source.
+* `getPhysicalFilename()` - returns the full path of the file on disk without any code block information.
```

Comment:
This may sound like we guarantee that the file exists on disk, while that doesn't have to be the case when the linting is run through [`lintText()`](https://eslint.org/docs/developer-guide/nodejs-api#-eslintlinttextcode-options), or on cli with the [--stdin](https://eslint.org/docs/user-guide/command-line-interface#using-stdin) option.

Reply:
I think this wording is okay. We are already saying it’s the “physical filename” which implies it exists on disk. I’d just suggest adding a note to clarify. 

```suggestion
* `getPhysicalFilename()` - when linting a file, returns the full path of the file on disk without any code block information; when linting text, the value passed to `—stdin-filename` or `<text>` if not specified. 
```

Reply:
Updated 👍🏻